cmd/configschema/yamlgen/yamlgen/README.md

```diff
@@ -0,0 +1,102 @@
+# Config Schema YAML Generator (alpha)
+
+This CLI application creates configschema YAML files for collector components
+where each YAML file describes the configuration fields of its corresponding
+component.
+
+## Operation
+
+There are two major modes of operation: one that create YAML files in each
+collector component directory and another that creates a zip archive comprising
+all the collector YAML files.
```

Comment:
Why do we need the zip file?

Reply:
The contents of the output zip file is for use by the collector configuration application that I've been working on, which needs this metadata. However, instead of writing a zip file, it could write dirs/files directly to the file system. In either case this is a different mode of operation that we'll need when we're generating config metadata for arbitrary components that may not have a locally available source dir.

Reply:
Ok, looks like you want these commands to be run by someone from the outside the repo, right? But why can't we enforce generation of cfg-metadata.yaml for every component with a makefile target and put them in the repo? In that case this running this command manually is not needed anymore, and zip file shouldn't be needed since your tool can always use pre-generated files, right?

We also have a CI job validating that all generated files are up to date https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/build-and-test.yml#L121

Reply:
I was thinking that this command could be used both outside this repo (e.g. in the Configurator (aka Otto) repo, wherever that ends up) and from within this repo as a makefile target as you mention, where we enforce generation of cfg-metadata.yaml for every component. 

Regarding not needing the the zip file mode, I am planning on replacing that run mode shortly with one that just writes to the filesystem (not to the collector source tree), but I think we'll still want that for other repos to access this data, because otherwise won't they have to checkout an entire collector source tree (and possibly multiple source trees for components that live in other distros) just to get the cfg-metadata.yaml files?